### DISCUSSION TITLE
Can't load the webpage when attempting to use pyppeteer

### DISCUSSION THREAD

**Discussion by liberty-rising** [idom-team/idom#706] [Problem]
**What was done to rule out other issues:**
I've tried using Pyppeteer to take snapshots of different webpages with no problem so the issue does not appear to be within its framework. I've also tried it on a locally hosted FASTAPI backend documentation page with no problem.

**Explanation of issue**
The issue arises when attempting to load up the main app which uses IDOM. The page's title tag and logo populate with no problem, however the issue appears to take place once the browser attempts to run the following script: 
```HTML
<script type="module">
      import { mount } from "./_snowpack/pkg/idom-app-react.js";
      mount(document.getElementById("app"));
</script>
```

Checking the developer console I get the following error:
`Uncaught SyntaxError: Unexpected token .`

Upon looking at the source of the issue, it is found in the following directory:
`http://localhost:8001/client/_snowpack/pkg/common/server-a6f83155.js`

Within this javascript file, the error is produced by the following function:
```JavaScript
const elementTransformCategories = {
  hasValue: (element) => ({
    value: element.value,
  }),
  hasCurrentTime: (element) => ({
    currentTime: element.currentTime,
  }),
  hasFiles: (element) => {
    if (element?.type === "file") {
      return {
        files: Array.from(element.files).map((file) => ({
          lastModified: file.lastModified,
          name: file.name,
          size: file.size,
          type: file.type,
        })),
      };
    } else {
      return {};
    }
  },
};
```

The exact issue shown within the script:
![image](https://user-images.githubusercontent.com/93674695/157429219-c66d88eb-08b5-4670-a5c2-313fe036f93f.png)

Once again, hovering over the red underlined code highlights the error as:
`Uncaught SyntaxError: Unexpected token .`

Any feedback would be greatly appreciated!

**Comment by rmorshea** [comment]
This looks like a browser compatibility issue with [this syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility). What web driver are you using?

**Reply by rmorshea** [reply]
It looks like Pyppeteer uses a pretty old version of Chromium by default: https://github.com/pyppeteer/pyppeteer/issues/291

There's a `PYPPETEER_CHROMIUM_REVISION` environment variable you can use to change that though.

**Reply by liberty-rising** [reply]
Ah looks like you were right, upon changing the version everything started working. Cheers!
